src/services/study/network-map.js

```diff
@@ -137,3 +146,102 @@ export function fetchAllCountries(studyUuid, currentNodeUuid) {
     console.debug(fetchCountriesUrl);
     return backendFetchJson(fetchCountriesUrl);
 }
+
+function createEquipmentIdentifierList(equipmentType, equipmentList) {
+    return {
+        type: 'IDENTIFIER_LIST',
+        equipmentType: equipmentType,
+        filterEquipmentsAttributes: equipmentList.map((eqId) => {
+            if (eqId?.id) {
+                return { equipmentID: eqId.id };
+            } else {
+                return { equipmentID: eqId };
+            }
+        }),
+    };
+}
+export async function createMapFilter(
+    filter,
+    distDir,
+    studyUuid,
+    currentNodeUuid,
+    networkMapref
+) {
+    let equipementList = [];
+    switch (filter.equipmentType) {
+        case EQUIPMENT_TYPES.SUBSTATION:
+            equipementList = createEquipmentIdentifierList(
+                filter.equipmentType,
+                networkMapref.current.getSelectedSubstation()
```

Comment:
Keep networkMapref near the component and pass selectedSubstation array to this function.